tests/Clients/GuzzleClientTest.php

```diff
@@ -72,8 +96,24 @@ public function testFormatPostParametersWithFiles()
         $this->assertEquals($response[5], ['name' => 'talents[]', 'contents' => 'stuff']);
         $this->assertEquals($response[6]['name'],       'resume');
         $this->assertEquals($response[6]['filename'],   'resume');
+        $this->assertEquals($response[7], ['name' => 'demographic_answers[]question_id', 'contents' => '1']);
+        $this->assertEquals($response[8], [
+          'name' => 'demographic_answers[]answer_options[][answer_option_id]',
+          'contents' => '100'
+        ]);
+        $this->assertEquals($response[9], [
+          'name' => 'demographic_answers[]answer_options[][answer_option_id]',
+          'contents' => '101'
+        ]);
+        $this->assertEquals($response[10], ['name' => 'educations[]school_name_id', 'contents' => '5417077']);
+        $this->assertEquals($response[11], ['name' => 'educations[]degree_id', 'contents' => '5494452']);
```

Comment:
How does this come through if you have more than one `educations`, though? If you have two elements you'd get `educations[]school_name_id` twice and I believe it'd only accept the second one, no?

Reply:
This is a mannerism of Rails Action Controllers documented here: https://edgeguides.rubyonrails.org/action_controller_overview.html#hash-and-array-parameters

When Rails encounters multiple `params` with the same name, they're treated as an array by the controller.

Indeed, all of these new client functionality depends on upon that behavior because this client encodes payloads as multipart forms, rather than JSON.